File: plugin-azure-server-arm/src/main/java/jetbrains/buildServer/clouds/azure/arm/connector/CredentialsAuthenticator.java
package jetbrains.buildServer.clouds.azure.arm.connector;

import okhttp3.*;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.io.IOException;

/**
 * Credentials request authenticator.
 */
public class CredentialsAuthenticator implements Authenticator {

    private final String myCredentials;

    public CredentialsAuthenticator(@NotNull final String username, @Nullable final String password) {
        myCredentials = Credentials.basic(username, password);
    }

    @Override
    public Request authenticate(Route route, Response response) throws IOException {
        return response.request().newBuilder()
                .header("Proxy-Authorization", myCredentials)
                .build();
    }
}

File: plugin-azure-server/src/main/kotlin/jetbrains/buildServer/clouds/azure/arm/connector/CredentialsAuthenticator.kt


package jetbrains.buildServer.clouds.azure.arm.connector

import okhttp3.Authenticator
import okhttp3.Credentials
import okhttp3.Request
import okhttp3.Response
import okhttp3.Route

/**
 * Credentials request authenticator.
 */
class CredentialsAuthenticator(username: String, password: String) : Authenticator {

    private val myCredentials: String = Credentials.basic(username, password)

    override fun authenticate(route: Route?, response: Response): Request {
        return response.request().newBuilder()
                .header("Proxy-Authorization", myCredentials)
                .build()
    }
}
